tests/e2e/tests/test_gateway.py

```diff
@@ -0,0 +1,118 @@
+#
+# Copyright 2024 Canonical, Ltd.
+#
+import logging
+from pathlib import Path
+
+import pytest
+from e2e_util import config, harness, util
+from e2e_util.config import MANIFESTS_DIR
+
+LOG = logging.getLogger(__name__)
+
+
+def test_gateway(h: harness.Harness, tmp_path: Path):
```

Comment:
I'm really starting to see much more similarity in each of these E2E tests.  They look more copy/paste than clearly defining what's being tested.  Once all the components are in place, i'd love to take another pass at these tests to make them more unified

Reply:
The similarity being there is true, however most of them are checking if pods are ready/deployed and making sure everything is settled. For example gateway and ingress basically provides the same thing functionality wise, so they both create some deployment and check if we can route requests. 

Most of these tests are verifying the simplest mode of operations, I believe our tests will grow more meaningful as components get different configuration options. +1 to revisiting our tests to make them more unified/clean.

Reply:
+1, let's revisit the tests _in the future_ to clean things up. I'm all fine with letting the tests be written independently at the moment, while we are still figuring out what they will look like.